### ISSUE TITLE
References to Async methods broken in versions after d7802f0

### ISSUE THREAD

**Issue by zakkatkk** [RicoSuter/NSwag#2208]
It would appear we are tied to a version of NSwag before a change to the BaseClientGenerator.cs in commit d7802f0, starting at line 125. "Async" is now stripped from the names of generated methods. Since development proceeded under previous versions of NSwag, several components incorrectly reference the generated methods after running the latest version of NSwag.

**Comment by RicoSuter** [comment]
Can you provide some samples?

**Comment by RicoSuter** [comment]
Is it about this?

https://github.com/RicoSuter/NSwag/commit/3a2ae3936cd6b4703a58e73808131671d66f2906

**Comment by zakkatkk** [comment]
Yes the issue appears related to the code in the commit you referenced, but issues appear to happen as the consequence of earlier changes. The steps to reproduce the issue with my project are as follows:
Install version of NSwag v12.2.3  
Generate services file. All changes are generated into services file. Working as normal.
Install version of NSwag v12.2.4
Generate services file. Async suffix is removed from all method names on service.

For example:
public async Task<HttpResponseMessage> CreateAdjustmentCodeAsync([FromBody] AdjustmentCodeRequestDto adjustmentCodeRequest) 
becomes
createAdjustmentCode(adjustmentCodeRequest: AdjustmentCodeRequestDto): Observable<ResponseModel>

**Comment by RicoSuter** [comment]
This is by design - in javascript/typescript the convention is not to use an async suffix.

**Comment by zakkatkk** [comment]
Though I don't agree with the naming convention on our client myself, the trouble is the change broke a ton of references in a legacy project. It seems like overriding the async renaming convention would best be a configuration. Otherwise as users with async references we are presented with the options of either changing every reference on the project or tying ourselves to a specific version of NSwag. Neither seems preferable.

**Comment by kurtosmate** [comment]
Same issue here.
Previously the NSwag generated method names same as the server endpoint and now it cuts the "Async" ending from the method name.

For example we have a method in server side: GetPaymentAsync()
The previously NSwag created a proxy with name: getPaymentAsync()
And now it generates with name: getPayment()

I'd like to refresh my fronted proxy to the server and now it caused tons of errors.

**Comment by HanaaGhasoub** [comment]
Same issue I faced now #2810

**Comment by brunobertechini** [comment]
Just my two cents: I believe this standard is correct (remove the async when generating typescript) but IMO it shouldn't be modified in a minor version since it is a breaking change (following semver). Perhaps this should have been done in a major bump? 12->13?

**Comment by Destidom** [comment]
Sorry for reviving such a old issue @RicoSuter, but we are currently facing a similar situation where a provider has given us an OpenAPI specification (OAS) that includes both 'Method' and 'MethodAsync'. 

Due to the fix for issue #2143 commit [3a2ae39](https://github.com/RicoSuter/NSwag/commit/3a2ae3936cd6b4703a58e73808131671d66f2906), NSwag now generates conflicting methods when the Async suffix is removed.

The main issue I think is that NSwag does not check for potential naming collisions after removing async from method names, thus resulting in having two Method with identical parameters and one with Task and and the other Task<object> as returning types. 
Thus the client will not build with the `'Error CS0111' : Type 'IMethod' already defines a member called 'MethodAsync' with the same parameter types.`

While I agree that async should not be part of method names or appear in an OAS, the specification does not enforce naming conventions. And as a consumer of such a OAS, it becomes very limiting when nswag apply a strict convention that a provider is not required to follow.

PR #4817 would resolve the naming collision by allowing consumers to decide how async naming should be handled in OAS files with unconventional method names, enabling valid client generation when both Method and MethodAsync exist.

Is there a particular reason PR #4817 was never merged? If needed, I can help resolve the merge conflicts, but Id like to understand the background first. if there is a fix in or workaround for this issue already that I have not found, if so I do apologize for reviving this issue.
